src/components/network/use-equipments.js

```diff
@@ -0,0 +1,29 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { loadEquipments } from 'redux/actions';
+
+export const useEquipments = (equipmentType) => {
+    const dispatch = useDispatch();
+    const equipments = useSelector(
+        (state) => state.spreadsheetNetwork[equipmentType.resource]
+    );
+    const studyUuid = useSelector((state) => state.studyUuid);
+    const currentNode = useSelector((state) => state.currentTreeNode);
+
+    const isFetching = equipments == null;
+
+    useEffect(() => {
+        if (equipments == null) {
+            Promise.all(
+                equipmentType.type.fetchers.map((fetcher) =>
+                    fetcher(studyUuid, currentNode.id)
+                )
+            ).then((results) => {
+                const equipments = [].concat(...results);
```

Comment:
Why this line ? `results` is already an array no ?

Reply:
After discussion, maybe use `.flat` instead

Reply:
fixed